pkg/controllers/managementuser/snapshotbackpopulate/snapshotbackpopulate.go

```diff
@@ -273,12 +302,31 @@ func (h *handler) populateUpstreamSnapshotFromDownstream(upstream *rkev1.ETCDSna
 		storage = Local
 	}
 
+	var (
+		definitiveName                  string
+		snapshotOriginalNameAnnotations map[string]string
+	)
+
+	base := strings.ToLower(InvalidKeyChars.ReplaceAllString(downstream.Spec.SnapshotName, "-"))
+	candidate := name.SafeConcatName(cluster.Name, base, string(storage))
+	if errs := validation.IsDNS1123Subdomain(candidate); len(errs) != 0 || len(candidate)+13 > validation.DNS1123SubdomainMaxLength {
+		safeSnapshotName := generateSafeSnapshotName(&downstream.Spec, downstream.Status.CreationTime.Time)
+		definitiveName = name.SafeConcatName(cluster.Name, safeSnapshotName)
+
+		snapshotOriginalNameAnnotations = map[string]string{
+			SnapshotOriginalNameKey: downstream.Spec.SnapshotName,
+			SnapshotNameStrategyKey: FallbackGeneratedStrategyName,
+		}
+	} else {
+		definitiveName = candidate
+		snapshotOriginalNameAnnotations = nil
+	}
```

Comment:
Honestly it might be a better approach to always set the `etcdsnapshot.rke.io/original-name` annotation (though I would rename it to `etcdsnapshot.rke.io/snapshot-name` or something), and omit the `etcdsnapshot.rke.io/name-strategy` key.

Something like:
```
name := generateSnapshotName(cluster, downstream)
...
upstream.Annotations[SnapshotOriginalNameKey] = downstream.Spec.SnapshotName
```

This way you are moving the complexity to the `generateSnapshotName` function, which should be much easier to test in isolation.

WDYT?